### ISSUE TITLE
Questions for ML4CO-KIDA

### ISSUE THREAD

**Issue by ihatebroccoli** [megvii-research/NeurIPS2021-ML4CO-KIDA#3]
Hello, I am master's student of A.I. course.

First, i'm really congratulations your ML4CO dual task winning.

I read your paper "ML4CO-KIDA: Knowledge Inheritance in Dataset Aggregation" recently. but i am beginner of this field(solving COP with ML), i have some questions for this paper.

**1**. Are there reasons why KIDA works better than EWA?(What i thought is, KIDA can calculate average policy evenly)

**2**. How can KIDA work well? I thought it will have some limitations, because it use strong branching as data which did bad performance on test case, and it trains with loss(strong branching, policy prediction)

**3**. in Figure 3: step 0~1000, there is big drop. is that coincidence? or representation of drop?(represent train accuracy at step 0)

![image](https://user-images.githubusercontent.com/10103856/158330735-66af2482-62b5-4f42-9df9-6615d30752a1.png)

**Comment by caozixuan** [comment]
Thank you for your attention to our work .

1. EWA aims at the same data set. It averages the parameters in different stages of training. Maybe the model after averaging is still very biased.

2. Strong branching is not completely ineffective, especially in the first and third problems. The model imitating strong branching is much better than the random model. If we can find when Strong Branching is a good decsion for current branching, the final performance of the model may be further improved.

3. Sorry, I didn't get what you said. Figure 3  reflects the accuracy curve obtained when the trained baseline model interacts with the actual solver. For example, if there are 20 steps in 0~100 steps are consistent with Strong Branching choices, the accuracy is 0.2 and there will be a point (0, 0.2) on the Figure. In the same way, (100, 0.15) represents the accuracy of 100~200 steps is 0.15. This figure shows that the accuracy of baseline method is low in actual interaction, and will decrease with the increase of the number of steps interacting with the solver

**Comment by ihatebroccoli** [comment]
Thanks for your answer!

Now i understand what i asked

1. Clearly understand

2. I think i misunderstood "strong branching is ineffective" , because in this graph, strong branching's performance was below KIDA and DAGGER_20. So can i define "**1)** Strong branching is good algorithm on Item placement problem **2)** The reason KIDA show better performance than SB is SB made good dataset during training data generation, but it could not make good performance because of dual degeneracy on test case(below graph)"?
![image](https://user-images.githubusercontent.com/68096644/158534439-c7535451-e913-4337-ae38-73e6a01a899e.png)

3. Clearly understand

**Comment by caozixuan** [comment]
2. In this figure, it does show a kind of failure of SB on this problem. But I don't think this means that SB performs the worst in the same number of interaction steps in the test set. This is only one of the cases, and the number of interaction steps does not reach the number of actual interaction steps (more than 100000 steps). The data distribution of the training set and the test set are consistent. So SB should not fail in the test set, but may fail in some specific cases.

4. What is certain is that DAGGER improves the generalization of the model, which can be reflected by the accuracy (the performance of the model in the training set and the test set tends to be consistent). However, KIDA improves the reward by averaging the parameters of different models. We are not sure whether this is related to generalization.

**Comment by ihatebroccoli** [comment]
Thanks for your answer! Hope everything is good for your future research
